src/index.js

```diff
@@ -122,6 +123,12 @@ class PeerId {
     return this._idB58String
   }
 
+  // return string representation from RFC 0001: https://github.com/libp2p/specs/pull/209
+  toCIDString () {
+    const cid = new CID(1, 'libp2p-key', this.id, 'base32')
+    return cid.toBaseEncodedString('base32')
```

Comment:
Given this is `Stage 1`, I was not sure if we should generate CIDString in constructor, or on-demand every time `toCIDString` is called.  Let me know if I should replace on-demand generation with  `return this._idCIDString`.

Reply:
Lazy generation would be good. I know it gets called a lot, especially for debug logs so not having to compute that every time would be great.

Also, what do you think about just making this `toString`? We could potentially overload the function to support other types, but I think it would be great to get away from using `to<format>String`. The goal is to use cids, so moving towards that being the default of `toString` feels like the right thing to do to me.

Reply:
Makes sense :+1: CIDv1 is self-describing (has multicodec which says "this is libp2p-key").
Added lazy-generation with memory and renamed to `toString` in https://github.com/libp2p/js-peer-id/pull/105/commits/60218d021a599239db2a10686be84033045b50fc

Reply:
A multicodec for libp2p-keys is a thing now? I need to catch up!

Reply:
@daviddias https://github.com/multiformats/multicodec/issues/130 :ok_hand: